src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -621,6 +621,42 @@ public static int typeDeclarationNameMatchingCount(String patternClass,
         return result;
     }
 
+    /**
+     * Calculates and returns the type declaration matching count when {@code classToBeMatched} is
+     * considered to be super class of an anonymous inner class.
+     *
+     * <p>
+     * Suppose our pattern class is {@code Main.j} and class to be matched is
+     * {@code Main.j.a.b} then type declaration name matching count would be calculated by
+     * comparing every character, and updating main counter when we hit "." or when it is the last
+     * character of the pattern class and certain conditions are met. This is done so that matching
+     * count is 6 instead of 5. This is due to the fact that pattern class can contain anonymous
+     * inner class object of a nested class which isn't true in case of extending classes as you
+     * can't extend nested classes.
+     * </p>
+     *
+     * @param patternTypeDeclaration type declaration against which the given type declaration has
+     *                               to be matched
+     * @param typeDeclarationToBeMatched type declaration to be matched
+     * @return type declaration matching count
+     */
+    public static int typeDeclarationMatchingCountAnonSuperClass(String patternTypeDeclaration,
```

Comment:
again, is this used somewhere else outside single check? If not, there is no need to put it to utils

Reply:
Will be used for fixing #11374

Reply:
https://github.com/checkstyle/checkstyle/pull/11468#discussion_r846579696